src/Glob.php

```diff
@@ -112,6 +112,84 @@ public static function glob($glob, $flags = 0)
     }
 
     /**
+     * Combines two globs/paths into another, stricter glob/path.
+     *
+     * @param string $exp1 Path or glob expression.
+     * @param string $exp2 Path or glob expression.
+     *
+     * @return null|string
+     */
+    public static function combine($exp1, $exp2)
+    {
+        $s1 = !self::isDynamic($exp1);
+        $s2 = !self::isDynamic($exp2);
+
+        // If both globs are static (paths), return the path if they are equal and null otherwise.
+        if ($s1 && $s2) {
+            return $exp1 === $exp2 ? $exp1 : null;
+        }
+
+        // If one glob is dynamic (wildcards) and the other static (path),
+        // return the path if it matches the glob and null otherwise.
+        if (!$s1 && $s2) {
+            return self::match($exp2, $exp1) ? $exp2 : null;
+        }
+        if ($s1 && !$s2) {
+            return self::match($exp1, $exp2) ? $exp1 : null;
+        }
+
+        // If both globs are dynamic, return the combined glob if they match each other and null otherwise.
+        if (!$s1 && !$s2) {
```

Comment:
Actually it should be possible to remove the if-statement entirely. It's always true, no?